CommonPackage.cmake

```diff
@@ -14,44 +25,181 @@ set(ENV{PKG_CONFIG_PATH}
+      set(COMMON_PACKAGE_COMPONENTS ON)
     else()
-      set(COMMON_PACKAGE_VERSION)
+      set(COMMON_PACKAGE_COMPONENTS)
     endif()
   endif()
 
-  if(COMMON_PACKAGE_QUIET)
-    set(COMMON_PACKAGE_FIND_QUIET "QUIET")
-  else()
-    list(FIND COMMON_PACKAGE_ARGS "QUIET" COMMON_PACKAGE_QUIET_POS)
-    if(COMMON_PACKAGE_QUIET_POS EQUAL -1)
-      set(COMMON_PACKAGE_FIND_QUIET)
-    else()
+  # OPT: only forward to find_package if not found or component find (tbd
+  # implement found_components check)
+  if((NOT ${Name}_FOUND AND NOT ${COMMON_PACKAGE_NAME}_FOUND) OR COMMON_PACKAGE_COMPONENTS)
+    set(COMMON_PACKAGE_VERSION)
+    if(COMMON_PACKAGE_ARGS)
+      list(GET COMMON_PACKAGE_ARGS 0 COMMON_PACKAGE_VERSION)
+      if(COMMON_PACKAGE_VERSION MATCHES "^[0-9.]+$") # is a version
+        set(COMMON_PACKAGE_VERSION ">=${COMMON_PACKAGE_VERSION}")
+      else()
+        set(COMMON_PACKAGE_VERSION)
+      endif()
+    endif()
+
+    if(COMMON_PACKAGE_QUIET)
       set(COMMON_PACKAGE_FIND_QUIET "QUIET")
+    else()
+      list(FIND COMMON_PACKAGE_ARGS "QUIET" COMMON_PACKAGE_QUIET_POS)
+      if(COMMON_PACKAGE_QUIET_POS EQUAL -1)
+        set(COMMON_PACKAGE_FIND_QUIET)
+      else()
+        set(COMMON_PACKAGE_FIND_QUIET "QUIET")
+      endif()
+    endif()
+
+    list(FIND COMMON_PACKAGE_ARGS "REQUIRED" COMMON_PACKAGE_REQUIRED_POS)
+    if(COMMON_PACKAGE_REQUIRED_POS EQUAL -1) # Optional find
+      find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
+      if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
+        pkg_check_modules(${Name} ${Name}${COMMON_PACKAGE_VERSION}
+          ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way
+      endif()
+    else() # required find
+      list(REMOVE_AT COMMON_PACKAGE_ARGS ${COMMON_PACKAGE_REQUIRED_POS})
+      find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
+      if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
+        pkg_check_modules(${Name} REQUIRED ${Name}${COMMON_PACKAGE_VERSION}
+          ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way (and fail if needed)
+      endif()
     endif()
   endif()
 
-  list(FIND COMMON_PACKAGE_ARGS "REQUIRED" COMMON_PACKAGE_REQUIRED_POS)
-  if(COMMON_PACKAGE_REQUIRED_POS EQUAL -1) # Optional find
-    find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
-    if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
-      pkg_check_modules(${Name} ${Name}${COMMON_PACKAGE_VERSION}
-        ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way
+  if(EXISTS ${PROJECT_SOURCE_DIR}/CMake/FindPackagesPost.cmake)
+    include(${PROJECT_SOURCE_DIR}/CMake/FindPackagesPost.cmake)
+  endif()
+
+  string(TOUPPER ${PROJECT_NAME} UPPER_PROJECT_NAME)
+  set(DEFDEP "${UPPER_PROJECT_NAME}_USE_${COMMON_PACKAGE_NAME}")
```

Comment:
move DEFDEF down to where it is used (line 115). if possible, find a more meaningful name :-)